Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,311 +14,403 @@
+        [NSException raise:NSInvalidArgumentException format:@"object can not be nil."];
+    }
+
+    id ret[] = { object };
+    return [self initWithObjects:ret count:1];
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithObjects:(id)firstObj, ... {
+    va_list argList;
+    va_start(argList, firstObj);
+    std::vector<id> flatArgs = ConvertVAListToVector((id)firstObj, argList);
+    va_end(argList);
+    return [self initWithObjects:flatArgs.data() count:flatArgs.size()];
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    if (objects == nil) {
+        [NSException raise:NSInvalidArgumentException format:@"objects can not be nil."];
+    }
+
+    if (self = [super init]) {
+        for (NSUInteger i = 0; i < count; ++i) {
+            [self _insertObject:objects[i]];
+        }
+    }
+    return self;
+}
+/**
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (orderedSet == nil) {
+        return [self init];
+    }
+
+    auto values = orderedSet->setContainer;
+
+    NSMutableArray* arrayObjs = [NSMutableArray array];
+
+    for (id item : values) {
+        [arrayObjs addObject:item];
+    }
+
+    return [self initWithArray:arrayObjs range:range copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithSet:set copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (set == nil) {
+        return [self init];
+    }
+
+    NSMutableArray* arrayObjs = [NSMutableArray array];
+
+    for (id item : set) {
+        [arrayObjs addObject:item];
+    }
+    return [self initWithArray:arrayObjs range:NSMakeRange(0, [arrayObjs count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)init {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [super init];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (BOOL)containsObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return mapContainer.find(object) != mapContainer.end();
```

Comment:
This is just going to compare pointers, isn't it? Not Objective-C object equality?